### PR TITLE
[BIO-6387-6392] added darwin core and embargo field

### PR DESCRIPTION
**Pull request by awset**
https://dev.azure.com/BIO-WA/CKAN%20Mod/_workitems/edit/6387/
https://dev.azure.com/BIO-WA/CKAN%20Mod/_workitems/edit/6392/

changes
- added darwin core format, OOTB ckan need to have 1 resource with this format to be created before it shows up in the dropdown, 
- darwin core also visible on the search facet.
- added embargo field and its validator
- added command to auto publish when the embargo data is equal or less than today
- added email notification for dataset will be auto published in the next 7 days.

image
image
image
image

### CODE REVIEW

File: ckanext/dbca/cli.py [hunk 1]
```diff
@@ -1,20 +1,110 @@
...
+                )
+                context = {u"user": site_user[u"name"]}
+                # set the dataset to Public
+                data_dict['private'] = False
+                # set the embargo date to null
+                data_dict.pop('embargo')
+                tk.get_action('package_update')(context, data_dict)
+                log.info(f"Dataset {dataset_name} is public")
+            except tk.NotAuthorized:
+                tk.abort(
```
Comment (MarkCalvert) [inline ckanext/dbca/cli.py]: Please remove.
Use tk.error_shout(error_msg)

File: ckanext/dbca/dbca_dataset.yaml [hunk 2]
```diff
@@ -56,6 +56,11 @@ dataset_fields:
   display_property: dcat:Dataset/dcat:landingPage
   display_snippet: link.html
 
+- field_name: embargo
+  label: Move to Public viewing on
+  preset: dbca_embargo_date
```
Comment (MarkCalvert) [inline ckanext/dbca/dbca_dataset.yaml]: Remove preset

File: ckanext/dbca/dbca_presets.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (MarkCalvert) [inline ckanext/dbca/dbca_presets.json]: Remove preset

File: ckanext/dbca/logic/action.py [hunk 4]
```diff
@@ -19,7 +26,45 @@ def dbca_get_sum(context, data_dict):
...
 
 
+@tk.side_effect_free
+def dbca_get_packages_to_be_published_or_notified(context, data_dict):
+    '''
+    Get all packages that need to be publshed or need to be notified.
+    Notification will be sent for package that will be published in the next 7 days.
+    '''
+    # Current UTC date
+    utc_now = datetime.datetime.utcnow().date()
```
Comment (MarkCalvert) [inline ckanext/dbca/logic/action.py]: Convert to Perth timezone for compares

**Review by MarkCalvert — CHANGES_REQUESTED** [review]

File: ckanext/dbca/cli.py (hunk 1 above)
Comment (awset) [reply ckanext/dbca/cli.py]: abort removed, added error_shout

File: ckanext/dbca/dbca_dataset.yaml (hunk 2 above)
Comment (awset) [reply ckanext/dbca/dbca_dataset.yaml]: preset removed

File: ckanext/dbca/dbca_presets.json (hunk 3 above)
Comment (awset) [reply ckanext/dbca/dbca_presets.json]: file removed

File: ckanext/dbca/logic/action.py (hunk 4 above)
Comment (awset) [reply ckanext/dbca/logic/action.py]: converted

**Review by MarkCalvert — APPROVED** [review]

**Pull request merged**